File: algorithm/src/main/java/baekjoon/p10205/Main.java
package baekjoon.p10205;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int k = sc.nextInt();

        for (int i = 1; i <= k; i++) {
            int solve = solve(sc.nextInt(), sc.next());
            System.out.println(String.format("Data Set %d:\n%d\n", i, solve));
        }
    }

    private static int solve(int h, String str) {
        int ret = h;
        for (char ch : str.toCharArray()) {
            switch (ch) {
                case 'b':
                    ret--;
                    break;
                case 'c':
                    ret++;
                    break;
            }
        }

        return ret;
    }

}

File: algorithm/src/main/java/baekjoon/p10205/Main.kt
package baekjoon.p10205

fun main(args: Array<String>) {
    repeat(readLine()!!.toInt()) {
        val inputs = List(2) { readLine()!! }
        val solve = solve(inputs[0].toInt(), inputs[1])
        println("Data Set ${it + 1}:\n$solve\n")
    }
}

private fun solve(h: Int, str: String): Int {
    return str.fold(h) { acc, c ->
        acc + when (c) {
            'b' -> -1
            'c' -> 1
            else -> 0
        }
    }
}